qsiprep/tests/test_utils_grouping.py

```diff
@@ -117,22 +188,402 @@ def test_get_entity_groups_without_multipartid(tmpdir):
+            'sub-01_dir-AP_run-2_dwi.nii.gz',
+            'sub-01_dir-PA_dwi.nii.gz',
+        ],
+    ]
+    check_expected(scan_groups, expected)
+
+
+def test_group_dwi_scans_with_complex_relpaths(tmpdir):
+    """Test the group_dwi_scans function.
+
+    This is the same as test_group_dwi_scans_with_complex_b0fields,
+    but with IntendedFors using relative paths instead of B0Field* fields.
+    """
+    bids_dir = tmpdir / 'test_group_dwi_scans_with_complex_relpaths'
+    dset_yaml = os.path.join(get_test_data_path(), 'skeleton_complex_relpaths.yml')
+    generate_bids_skeleton(str(bids_dir), dset_yaml)
+    layout = BIDSLayout(str(bids_dir))
+    subject_data = {'dwi': layout.get(suffix='dwi', extension='nii.gz', return_type='file')}
+    scan_groups, _ = grouping.group_dwi_scans(
+        layout=layout,
+        subject_data=subject_data,
+        using_fsl=True,
+        combine_scans=True,
+        ignore_fieldmaps=False,
+        concatenate_distortion_groups=True,
+    )
+    expected = [
+        {
+            'concatenated_bids_name': 'sub-01',
+            'dwi_series': [
+                'sub-01_dir-PA_dwi.nii.gz',
+            ],
+            'dwi_series_pedir': 'j',
+            'fieldmap_info': {
+                'epi': [
+                    'sub-01_dir-AP_epi.nii.gz',
+                    'sub-01_dir-PA_epi.nii.gz',
+                ],
+                'rpe_series': [
+                    'sub-01_dir-AP_run-1_dwi.nii.gz',
+                    'sub-01_dir-AP_run-2_dwi.nii.gz',
+                ],
+                'suffix': 'rpe_series',
+            },
+        },
+    ]
+    check_expected(scan_groups, expected)
+
+    scan_groups, _ = grouping.group_dwi_scans(
+        layout=layout,
+        subject_data=subject_data,
+        using_fsl=False,
+        combine_scans=True,
+        ignore_fieldmaps=False,
+        concatenate_distortion_groups=False,
+    )
+    expected = [
+        {
+            'concatenated_bids_name': 'sub-01_dir-AP',
+            'dwi_series': [
+                'sub-01_dir-AP_run-1_dwi.nii.gz',
+                'sub-01_dir-AP_run-2_dwi.nii.gz',
+            ],
+            'dwi_series_pedir': 'j-',
+            'fieldmap_info': {
+                'epi': ['sub-01_dir-PA_epi.nii.gz'],
+                'suffix': 'epi',
+            },
+        },
+        {
+            'concatenated_bids_name': 'sub-01_dir-PA',
+            'dwi_series': [
+                'sub-01_dir-PA_dwi.nii.gz',
+            ],
+            'dwi_series_pedir': 'j',
+            'fieldmap_info': {
+                'epi': ['sub-01_dir-AP_epi.nii.gz'],
+                'suffix': 'epi',
+            },
+        },
+    ]
+    check_expected(scan_groups, expected)
+
+    scan_groups, _ = grouping.group_dwi_scans(
+        layout=layout,
+        subject_data=subject_data,
+        using_fsl=True,
+        combine_scans=False,
+        ignore_fieldmaps=False,
+        concatenate_distortion_groups=False,
+    )
+    expected = [
```

Comment:
@mattcieslak: This one is clearly an error. Amelie probably used this option recently, and it worked at the time so this might just be weird.